book/src/developers/quick_setup.md

```diff
@@ -115,6 +115,92 @@ cargo run -- --help
 cargo run
 ```
 
+## Building on MacOS Based Systems
+
+### Prerequisites
+- [Rust](https://www.rust-lang.org/) installation
+- Xcode Command Line Tools
+
+### Building, Testing, and Running
+
+Note: If you use a VPN, you should disable it before running Trin.
+
+Install Xcode Command Line Tools (MacOS):
+
+```sh
+xcode-select --install
+```
+
+Environment variables:
+
+```sh
+# Optional
+export RUST_LOG=<error/warn/info/debug/trace>
+export TRIN_DATA_PATH=<path-to-data-directory>
+```
+
+Install Rust dependencies (MacOS):
+
+```sh
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs/ | sh -s -- --default-toolchain nightly -y && \
+echo 'export PATH="${HOME}/.cargo/bin:${PATH}"' >> $HOME/.bashrc && \
```

Comment:
this won't work for users using Zsh instead of Bash since this should be appended to $HOME/.zshrc instead in that case for it to take effect automatically if the user opens another terminal window, otherwise they'd manually have to run `source $HOME/.bashrc`. so this would be better in a script that automatically sets it up for macOS users